src/components/create-courses/create-course.jsx

```diff
@@ -0,0 +1,181 @@
+import React, {useEffect, useState} from 'react';
+import { Container, Button, Input, Label, Form, Row, Col } from "reactstrap";
+import toHoursAndMinutes from "../../common/to-hours-and-minutes";
+import AuthorsLists from "./component/authors-lists";
+import AddNewAuthor from "./component/create-new-author";
+import { v4 as uuidv4 } from 'uuid';
+
+function CreateCourse({authors, addAuthor, addCourse, goToCourses}) {
+
+	// AuthorsLists states and functionality
+	const [authorsList, setAuthorsList] = useState(authors);
+
+	useEffect( () => {
+		setAuthorsList(authors.map(author => ({ ...author, selected: false })))
+	}, [authors])
+
+	//"Add New Author" functionality
+	function addNewAuthorToList(newAuthor) {
+		newAuthor !== undefined ? addAuthor(newAuthor) : console.log('newAuthor false',newAuthor)
+	}
+
+	// Form States, data and functionality
+	const [formState, setFormState] = useState({
+		id: '',
+		title: '',
+		description: '',
+		creationDate: '',
+		duration: 0,
+		authors: []
+	})
+
+	const handleChange = (evnt) => {
+		setFormState((prev) => ({
+			...prev,
+			[evnt.target.name]: evnt.target.value
+		}));
+	};
+
+	function getAuthorsId() {
+		const authorsId = []
+		for(let i = 0; i < authorsList.length; i++) {
+			if(authorsList[i].selected){
+				let id = authorsList[i].id
+				authorsId.push(id)
+			}
+		}
+		return authorsId
+	}
+
+	function validateCourseFormData(course) {
+		if( course.title === '') {
+			alert('Field "Title" is not filled in!')
+			return false
+		}
+		if( course.description === '') {
+			alert('Field "Description" is not filled in!')
+			return false
+		}
+		if( course.authors.length < 2) {
+			alert('The course must include at least two authors!')
+			return false
+		}
+		if( course.duration <= 0) {
+			alert('Field "Duration" is not filled in!')
+			return false
+		}
+		return true
+	}
+
+	function createNewCourse() {
+		const course = formState;
+		course.id= uuidv4();
```

Comment:
Just try to avoid mutations, the better way is spread operator

`return ({
...formState,
id: uuidv4(),
authors:  authorsList.map(a => ....)
})`

Reply:
Done!